### PR TITLE
Remove legacy pubsub for all agents

### PR DESCRIPTION
**Pull request by archishou**
Removed usages of legacy pubsub for all other agents.

### CODE REVIEW

**Comment by deitch** [comment]
yetus is unhappy, although I am unsure why.

**Comment by archishou** [comment]
All green!

File: pkg/pillar/cmd/baseosmgr/baseosmgr.go [hunk 1]
```diff
@@ -548,10 +581,14 @@ func initializeNodeAgentHandles(ctx *baseOsMgrContext) {
...
 }
 
-func initializeZedagentHandles(ctx *baseOsMgrContext) {
+func initializeZedagentHandles(ps *pubsub.PubSub, ctx *baseOsMgrContext) {
 	// Look for BaseOsConfig , from zedagent
-	subBaseOsConfig, err := pubsublegacy.Subscribe("zedagent",
-		types.BaseOsConfig{}, false, ctx, &pubsub.SubscriptionOptions{
+	subBaseOsConfig, err := ps.NewSubscription(
+		pubsub.SubscriptionOptions{
+			AgentName:     "nodeagent",
```
Comment (eriknordmark) [inline pkg/pillar/cmd/baseosmgr/baseosmgr.go]: @archishou Can you redo this in a way where the agentname is correct for all the changes?
Shouldn't depend on a code reviewer and/or testing to get the basic transformation correct.

FWIW I tend to do emacs macros to automate transformations as a way to ensure it is correct. You might prefer other tools. But manual edits is certainly the wrong approach to apply code transformations like this.

File: pkg/pillar/cmd/baseosmgr/baseosmgr.go [hunk 2]
```diff
@@ -548,10 +581,14 @@ func initializeNodeAgentHandles(ctx *baseOsMgrContext) {
...
+			TopicImpl:     types.ZbootConfig{},
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/baseosmgr/baseosmgr.go]: This is also incorrect - Copy Paste Error

File: pkg/pillar/cmd/domainmgr/domainmgr.go [hunk 3]
```diff
@@ -261,8 +281,12 @@ func Run(ps *pubsub.PubSub) {
 	domainCtx.subGlobalConfig = subGlobalConfig
 	subGlobalConfig.Activate()
 
-	subDeviceNetworkStatus, err := pubsublegacy.Subscribe("nim",
-		types.DeviceNetworkStatus{}, false, &domainCtx, &pubsub.SubscriptionOptions{
+	subDeviceNetworkStatus, err := ps.NewSubscription(
+		pubsub.SubscriptionOptions{
+			AgentName:     "",
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/domainmgr/domainmgr.go]: This should be nim - Copy paste error?

File: pkg/pillar/cmd/ledmanager/ledmanager.go [hunk 4]
```diff
@@ -177,43 +176,52 @@ func Run(ps *pubsub.PubSub) {
...
-	subLedBlinkCounter, err := pubsublegacy.Subscribe("", types.LedBlinkCounter{},
-		false, &ctx, &pubsub.SubscriptionOptions{
-			CreateHandler: handleLedBlinkModify,
-			ModifyHandler: handleLedBlinkModify,
-			DeleteHandler: handleLedBlinkDelete,
-			WarningTime:   warningTime,
-			ErrorTime:     errorTime,
-		})
+	subLedBlinkCounter, err := ps.NewSubscription(pubsub.SubscriptionOptions{
+		AgentName:     "zedmanager",
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/ledmanager/ledmanager.go]: Agent should be "" -- Copy paste Error??

**Comment by deitch** [comment]
Why did you close this?

**Comment by archishou** [comment]
Going to finish it and then open when ready for review

**Comment by deitch** [comment]
Excited to see this come to fruition.

**Comment by deitch** [comment]
Needs a rebase, as @eriknordmark said, but looks good other than that.

**Comment by deitch** [comment]
I cannot shake the feeling that there is a way to make the initialization of each agent cleaner and easier. It doesn't affect this PR in the least, but I will try to spend some time cogitating on it.

File: pkg/pillar/cmd/nim/nim.go [hunk 5]
```diff
@@ -117,37 +116,50 @@ func Run(ps *pubsub.PubSub) {
...
 	}
 	pubDevicePortConfig.ClearRestarted()
 
-	pubDevicePortConfigList, err := pubsublegacy.PublishPersistent(agentName,
-		types.DevicePortConfigList{})
+	pubDevicePortConfigList, err := ps.NewPublication(
+		pubsub.PublicationOptions{
+			AgentName:  agentName,
+			Persistent: true,
+			TopicType:  types.DevicePortConfig{},
```
Comment (eriknordmark) [inline pkg/pillar/cmd/nim/nim.go]: That's why nim fails/hangs; the type should be DevicePortConfigList!

File: pkg/pillar/cmd/logmanager/logmanager.go [hunk 6]
```diff
@@ -192,29 +195,35 @@ func Run(ps *pubsub.PubSub) {
...
-			ModifyHandler: handleDomainStatusModify,
-			DeleteHandler: handleDomainStatusDelete,
-			WarningTime:   warningTime,
-			ErrorTime:     errorTime,
-		})
+	subDomainStatus, err := ps.NewSubscription(pubsub.SubscriptionOptions{
+		AgentName:     "domainmgr",
+		TopicImpl:     types.DomainStatus{},
+		Activate:      false,
+		Ctx:           logmanagerCtx,
```
Comment (eriknordmark) [inline pkg/pillar/cmd/logmanager/logmanager.go]: Missing & results in panic with 
2020-03-12T14:28:18.490975+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;panic: interface conversion: interface {} is logmanager.DNSContext, not *logmanager.DNSContext
2020-03-12T14:28:18.683559+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;github.com/lf-edge/eve/pkg/pillar/cmd/logmanager.handleDNSModify(0x1b33b80, 0xc00013e9f0, 0xc00044901a, 0x6, 0x1b02800, 0xc00000e020)
2020-03-12T14:28:18.735955+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;	/pillar/cmd/logmanager/logmanager.go:418 +0x5a5
2020-03-12T14:28:19.028682+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;github.com/lf-edge/eve/pkg/pillar/cmd/logmanager.Run(0xc000438000)
2020-03-12T14:28:19.075908+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;	/pillar/cmd/logmanager/logmanager.go:273 +0x144d

File: pkg/pillar/cmd/logmanager/logmanager.go [hunk 7]
```diff
@@ -225,14 +234,17 @@ func Run(ps *pubsub.PubSub) {
...
-			ModifyHandler: handleDNSModify,
-			DeleteHandler: handleDNSDelete,
-			WarningTime:   warningTime,
-			ErrorTime:     errorTime,
-		})
+	subDeviceNetworkStatus, err := ps.NewSubscription(pubsub.SubscriptionOptions{
+		AgentName:     "nim",
+		TopicImpl:     types.DeviceNetworkStatus{},
+		Activate:      false,
+		Ctx:           DNSctx,
```
Comment (eriknordmark) [inline pkg/pillar/cmd/logmanager/logmanager.go]: Sorry it was this one

**Comment by archishou** [comment]
For the usages in utils, I started with ledmanagerutils.go and found that so many different functions and files used the pubsub instance, it would be really messy to change all the function definitions to include the ps *pubsub.PubSub. Instead, would it be ok to use the default pubsub without having to get the one from the agent? Or should all the function definitions be changed?

**Comment by eriknordmark** [comment]
I didn't realize the utils would be that convoluted. Let finish the agents and do a separate PR later for the utils.

**Comment by archishou** [comment]
Ok, I'll remove the last commit then.

**Pull request merged**